catalog/ui/catalog-ui-search/src/main/webapp/js/model/Workspace.js

```diff
@@ -213,30 +238,49 @@ define([
             duplicateWorkspace: function(workspace){
                 this.create(_.omit(workspace.toJSON(), 'id', 'owner', 'metacard.sharing'));
             },
-            saveLocal: function () {
+            saveAll: function(){
+                this.forEach(function(workspace){
+                    if (!workspace.isSaved()){
+                        workspace.save();
+                    }
+                });
+            },
+            saveLocalWorkspaces: function () {
                 var localWorkspaces = this.chain()
                     .filter(function (workspace) {
                         return workspace.get('localStorage');
                     })
-                    .map(function (workspace) {
-                        return workspace.toJSON();
-                    })
+                    .reduce(function (blob, workspace) {
+                        blob[workspace.get('id')] = workspace.toJSON();
+                        return blob;
+                    }, {})
                     .value();
 
                 window.localStorage.setItem('workspaces', JSON.stringify(localWorkspaces));
             },
+            convertOldFormat: function(localWorkspaceJSON){
```

Comment:
Not thinking we'll be changing from the current method of storing them, but we can name it something like convertArrayStoredWorkspacesToMapStoredWorkspaces.